release/models/system/openconfig-system-logging.yang

```diff
@@ -433,6 +435,14 @@ module openconfig-system-logging {
         "Sets the destination port number for syslog UDP messages to
         the server.  The default for syslog is 514.";
     }
+
+    leaf facility-override {
+      type identityref {
+        base SYSLOG_FACILITY;
+      }
+      description
+        "Override the syslog facility level to the log server";
```

Comment:
What is the behavior you expect when this leaf is specified?  I'm a little confused by "facility level" here -- do you mean to override the severity? (the identityref implies override of the facility though)

Reply:
No - this is to override the facility.  e.g. say all internal processes are logging at facility levels local5/local6/local7 and the operator chooses to override all towards the remote-server to local0.

This feature exists on various major implementations (see below)

```
user@junos# set system syslog host 1.1.1.1 facility-override ?
Possible completions:
  authorization        Authorization system
  daemon               Various system processes
  ftp                  FTP process
  kernel               Kernel
  local0               Local logging option number 0
  local1               Local logging option number 1
  local2               Local logging option number 2
  local3               Local logging option number 3
  local4               Local logging option number 4
  local5               Local logging option number 5
  local6               Local logging option number 6
  local7               Local logging option number 7
  user                 User processes
``` 

```
eos(config)#logging facility ?
  auth    Authorization messages    (facility=4)
  cron    Clock daemon              (facility=15)
  daemon  System daemons            (facility=3)
  kern    Kernel messages           (facility=0)
  local0  Local use 0               (facility=16)
  local1  Local use 1               (facility=17)
  local2  Local use 2               (facility=18)
  local3  Local use 3               (facility=19)
  local4  Local use 4               (facility=20)
  local5  Local use 5               (facility=21)
  local6  Local use 6               (facility=22)
  local7  Local use 7               (facility=23)
  lpr     Line printer subsystem    (facility=6)
  mail    Mail system               (facility=2)
  news    Network news subsystem    (facility=7)
  sys10   System use 10             (facility=10)
  sys11   System use 11             (facility=11)
  sys12   System use 12             (facility=12)
  sys13   System use 13             (facility=13)
  sys14   System use 14             (facility=14)
  sys9    System use 9              (facility=9)
  syslog  Internal syslogd messages (facility=5)
  user    User-level messages       (facility=1)
  uucp    UUCP subsystem            (facility=8)
  <0-23>  Syslog facility value
```

```
RP/0/RP0/CPU0:ios(config)#logging facility ?
  auth    Authorization system
  cron    Cron/at facility
  daemon  System daemons
  kern    Kernel
  local0  Local use
  local1  Local use
  local2  Local use
  local3  Local use
  local4  Local use
  local5  Local use
  local6  Local use
  local7  Local use
  lpr      Line printer system
  mail    Mail system
  news    USENET news
  sys10   System use
  sys11   System use
  sys12   System use
  sys13   System use
  sys14   System use
  sys9    System use
  syslog  Syslog itself
  user    User process
  uucp    Unix-to-Unix copy system
  <cr>    
```